subsys/esb/esb.c

```diff
@@ -1051,13 +1077,38 @@ int esb_init(const struct esb_config *config)
 	sys_timer_init();
 	ppi_init();
 
+	#if IS_ENABLED(CONFIG_ESB_DYNAMIC_INTERRUPTS)
+
+	/* Ensure IRQs are disabled before attaching. */
+	irq_disable(RADIO_IRQn);
```

Comment:
Please group all irq_disable call into static function and use it here and in the `esb_disable` function

Reply:
Done